source/retryable-reads/tests/unified/handshakeError.yml

```diff
@@ -0,0 +1,76 @@
+description: "retryable reads handshake failures"
+
+schemaVersion: "1.3"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    auth: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, connectionCheckOutStartedEvent]
+      uriOptions: { retryReads: true, retryWrites: true, maxPoolSize: 5 }
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-handshake-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  - description: "find succeeds after retryable handshake network error"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              # use saslContinue here to avoid SDAM errors
+              # this failPoint itself will create a usable connection in the connection pool
+              # so we run a dropCollection (that also fails) in order to discard the connection
+              # before testing our read operation "find"
+              failCommands: [saslContinue, dropCollection]
+              closeConnection: true
+
+      - name: dropCollection
+        object: *database0
+        arguments:
+          collection: doesNotMatter
```

Comment:
If `dropCollection` doesn't actually matter, would it be preferable to use something more benign like a `ping` command?

Reply:
@ShaneHarvey Switching to ping seems like it would be a fine equivalent, anything specific about drop's behavior? we just need to run something that will have the connection get shut upon running it

Reply:
Yes it can be any command the ends of clearing the pool. Switching to ping SGTM.